### PR TITLE
feat(rpc): extra validation send_raw_transaction

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.2 day

Resolves: #1264 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Add validation of the transaction's gas limit, max priority fee and max fee.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by tcoratger** [comment]
About the gas fee test that fails on the retry, we should probably modify places in the codebase where we have hardcoded stuffs for test like:

https://github.com/kkrt-labs/kakarot-rpc/blob/a93f559f113e70500890ccfa79107529d490584b/src/test_utils/eoa.rs#L226

which doesn't respect your validation condition, wdyt?

**Comment by greged93** [comment]
trying rn but quite complicated

File: src/eth_provider/database/ethereum.rs [hunk 1]
```diff
@@ -105,6 +107,14 @@ pub trait EthereumBlockStore {
 
 #[async_trait]
 impl EthereumBlockStore for Database {
+    async fn latest_header(&self) -> Result<Option<Header>, EthApiError> {
+        let filter = EthDatabaseFilterBuilder::<filter::Header>::default().build();
+        Ok(self
+            .get_one::<StoredHeader>(filter, Some(doc! { "number": -1 }))
```
Comment (tcoratger) [inline src/eth_provider/database/ethereum.rs:113]: Here why not using our FindOpts wrapper with a `with_sort` function:

https://github.com/mongodb/mongo-rust-driver/blob/b781af26dfb17fe62823a866a025de9fb102e0b3/src/coll/options.rs#L845

This probably also requires to modify the `get_one` function signature and adding one function to our wrapper

**Comment by tcoratger** [comment]
> trying rn but quite complicated

I imagine that this validation also breaks some hive tests no?

**Comment by greged93** [comment]
I think the hive tests failed because of block number, made a mistake when modifying but should be ok (tm)

File: src/eth_provider/database/ethereum.rs (hunk 1 above)
Comment (tcoratger) [reply src/eth_provider/database/ethereum.rs:113]: Yes but I think you should be able to build from https://github.com/kkrt-labs/kakarot-rpc/blob/4003bd0518102f7cd971d86afa143f86253f6e07/src/eth_provider/database/mod.rs#L43

**Review by tcoratger — APPROVED** [review]

**Pull request merged**